src/main/routes/port-forward-route.ts

```diff
@@ -118,33 +125,73 @@ class PortForward {
 
 export class PortForwardRoute {
   static async routePortForward(request: LensApiRequest) {
+    const { params, response, cluster } = request;
+    const { namespace, port, forwardPort, resourceType, resourceName } = params;
+
+    try {
+      let portForward = PortForward.getPortforward({
+        clusterId: cluster.id, kind: resourceType, name: resourceName,
+        namespace, port, forwardPort,
+      });
+
+      if (!portForward) {
+        logger.info(`Creating a new port-forward ${namespace}/${resourceType}/${resourceName}:${port}`);
+        portForward = new PortForward(await cluster.getProxyKubeconfigPath(), {
+          clusterId: cluster.id,
+          kind: resourceType,
+          namespace,
+          name: resourceName,
+          port,
+          forwardPort
+        });
+
+        const started = await portForward.start();
+
+        if (!started) {
+          return respondJson(response, {
+            message: `Failed to forward port ${port} to ${forwardPort}`
+          }, 400);
+        }
+      }
+
+      portForward.open();
+      respondJson(response, {port: portForward.forwardPort});
+    } catch (e) {
+      return respondJson(response, {
+        message: `Failed to forward port ${port} to ${forwardPort}`
+      }, 400);
+    }
+  }
+
+  static async routeCurrentPortForward(request: LensApiRequest) {
     const { params, response, cluster} = request;
-    const { namespace, port, resourceType, resourceName } = params;
-    let portForward = PortForward.getPortforward({
+    const { namespace, port, forwardPort, resourceType, resourceName } = params;
+
+    const portForward = PortForward.getPortforward({
       clusterId: cluster.id, kind: resourceType, name: resourceName,
-      namespace, port
+      namespace, port, forwardPort
     });
 
-    if (!portForward) {
-      logger.info(`Creating a new port-forward ${namespace}/${resourceType}/${resourceName}:${port}`);
-      portForward = new PortForward(await cluster.getProxyKubeconfigPath(), {
-        clusterId: cluster.id,
-        kind: resourceType,
-        namespace,
-        name: resourceName,
-        port,
-      });
-      const started = await portForward.start();
+    let forwardedPort = -1;
 
-      if (!started) {
-        return respondJson(response, {
-          message: "Failed to open port-forward"
-        }, 400);
-      }
+    if (portForward) {
+      forwardedPort = Number(portForward.forwardPort);
     }
```

Comment:
This can be simplified:

```suggestion
    const forwardedPort = Number(portForward?.forwardPort ?? -1);
```